[docs]: add SCALE codec troubleshooting section (#428) (#429)
hyperledger/iroha-2-docs: src/guide/troubleshooting/configuration-issues.md

These sorts of subtle mistakes can be avoided, for example, by
deserialising directly from string literals, or by generating a fresh
key-pair in places where it makes sense.

## SCALE codec errors

You may encounter SCALE codec errors when there's a discrepancy

> between the data structure on the client and the peer sides.

Reviewer: I would rephrase. SCALE codec errors appear when the versions of the data structures differ between the client and the peer sides.

> between how the data is structured on the client and the peer sides.

Reviewer 2: ```suggestion
between the data structure on the client and the peer sides.
```

It's easy to confuse the builds in the `debug` and `release` directories,
and keep using an old build due to this.

> In case you're using `iroha_client_cli` and rebuild it inside the Iroha

Reviewer 2: ```suggestion
In case you're using the `iroha_client_cli` and rebuild it inside the Iroha
```

> project directory root:

Reviewer 2: ```suggestion
project directory root, use these commands:
```

* `rm ./target/debug/iroha_client_cli ./target/release/iroha_client_cli`

> * `cargo build -p iroha_client_cli`

Reviewer 3: this should be a code block, not a list

> * `cargo build -p iroha_client_cli`

Reviewer: ```suggestion
* `cargo build -p iroha_client_cli --release`
```

Alternatively, please compare the output of

> `/target/debug/iroha_client_cli --version` with a version in the Iroha log.

Reviewer: ```suggestion
`cargo run -p iroha_client_cli --release -- --version` with a version in the Iroha log.
```

> You need to view the part at which the Iroha peer starts.

Reviewer 2: ```suggestion
You need to view the part where the Iroha peer starts.
```

> You need to view the part where the Iroha peer starts.

Reviewer: > You need to view the part where the Iroha peer starts.

Which part? Provide an example.

This will allow you to see whether the versions match.

> If you are using one of our SDKs, please make sure to update it.

Reviewer 2: ```suggestion
If you are using one of our SDKs, please make sure it is up to date.
```

> If you are using one of our SDKs, please make sure it is up to date.

Reviewer: It should be not necessarily up to date, but the version of SDK should be compatible with the version of Iroha. In order to check it, it is generally recommended to check for the version in the SDKs readme.
